airbyte-integrations/connectors/source-mailchimp/source_mailchimp/streams.py

```diff
@@ -0,0 +1,178 @@
+        Defining a cursor field indicates that a stream is incremental, so any incremental stream must extend this class
+        and define a cursor field.
+        """
+        pass
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        """
+        Return the latest state by comparing the cursor value in the latest record with the stream's most recent state object
+        and returning an updated state object.
+        """
+        latest_state = latest_record.get(self.cursor_field)
+        current_state = current_stream_state.get(self.cursor_field) or latest_state
+        return {self.cursor_field: max(latest_state, current_state)}
+
+    def request_params(self, stream_state=None, **kwargs):
+
+        stream_state = stream_state or {}
+        params = super().request_params(stream_state=stream_state, **kwargs)
+        default_params = {"sort_field": self.cursor_field, "sort_dir": "ASC"}
+        since_value = stream_state.get(self.cursor_field)
+        if since_value:
+            default_params[f"since_{self.cursor_field}"] = since_value
+        params.update(default_params)
+        return params
+
+
+class Lists(IncrementalMailChimpStream):
+    cursor_field = "date_created"
+    data_field = "lists"
+
+    def path(self, **kwargs) -> str:
+        return "lists"
+
+
+class Campaigns(IncrementalMailChimpStream):
+    cursor_field = "create_time"
+    data_field = "campaigns"
+
+    def path(self, **kwargs) -> str:
+        return "campaigns"
+
+
+class EmailActivity(IncrementalMailChimpStream):
+    cursor_field = "timestamp"
+    name = "Email_activity"
+    data_field = "emails"
+
+    def stream_slices(self, **kwargs):
+        campaign_stream = Campaigns(authenticator=self.authenticator)
+        for campaign in campaign_stream.read_records(sync_mode=SyncMode.full_refresh):
+            yield {"campaign_id": campaign["id"]}
+
+    def path(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
+        self.campaign_id = stream_slice["campaign_id"]
+        return f"reports/{self.campaign_id}/email-activity"
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        """
+        Return the latest state by comparing the campaign_id and cursor value in the latest record with the stream's most recent state object
+        and returning an updated state object.
+        """
+
+        latest_cursor_value = latest_record.get(self.cursor_field)
+        current_state = current_stream_state.get(self.campaign_id) if current_stream_state else None
+        if current_state:
+            current_state = current_state.get(self.cursor_field)
+        current_state_value = current_state or latest_cursor_value
+        max_value = max(current_state_value, latest_cursor_value)
+        output = {self.campaign_id: {self.cursor_field: max_value}}
```

Comment:
will not work because we always rewrite campaigns that we already have, we should update it. Also record container campaing_id field, please use it.

the resulting state should looks like:
```python
{
   "123": {
      "since": "2010-10-01 03:21:11.33"
   },
   "456": {
      "since": "2020-10-01 03:21:11.33"
   },
}
```

Reply:
thats why this method is called `update`